src/lib/random/xoroshiro.h

```diff
@@ -30,12 +30,12 @@ __extension__ ({							\
 })
 
 // FIXME: this is a very poor way to seed the generator
-#define random_init(rng_s, llid)					\
+#define random_init(rng_s, llid, seed)					\
 __extension__ ({							\
-	(rng_s)[0] = (llid + 1) * UINT64_C(16232384076195101791);	\
-	(rng_s)[1] = (llid + 1) * UINT64_C(13983006573105492179);	\
-	(rng_s)[2] = (llid + 1) * UINT64_C(10204677566545858177);	\
-	(rng_s)[3] = (llid + 1) * UINT64_C(14539058011249359317);	\
+	(rng_s)[0] = (llid + 1) * UINT64_C(16232384076195101791) ^ seed;\
```

Comment:
I'm not 100% sure this is not introducing a bias in the sequence. Could you confirm this?

Reply:
Assuming the base seeding procedure generates a sufficiently random initial state, the xor operation doesn't introduce biases (think of it as a one-time pad). The actual problem is that the base seeding procedure is not that good to begin with (if you want, we talk about the problem more extensively). Still, we are dealing with 256 bit of state, so even a sub-optimal seeding procedure would still guarantee very well spread output sequences for the scale of our current simulations.

Reply:
We should talk about this in person. I see some problems with this kind of seeding, but I'm very possibly wrong!

Reply:
Let's keep it as is for now.